Source/Applications/ApplicationArtifactIdentifierStringConverter.cs

```diff
@@ -85,7 +89,8 @@ public IApplicationArtifactIdentifier FromString(string identifierAsString)
         {
             ValidateIdentifierString(identifierAsString);
 
-            var regex = new Regex(@"(\w+)#(?:([\w]+)[.]*)+-([\w]+)\+([\w]+)\|([\w]+)");
+            //TODO: Add regex capture group for generation
+            var regex = new Regex(@"(\w+)#(?:([\w]+)[.]*)+-([\w]+)\+([\w]+)\+([0-9]+)\|([\w]+)");
```

Comment:
Suggestion; for consistency and less error prone - should we change from hardcoded separators in the string to using the constants directly in the string?  Make it something like this:

```csharp
var regex = new Regex(
   @"(\w+)"+ApplicationSeparator+
   @"(?:([\w]+)[.]*)"+ ApplicationLocationSeparator+
   ...
```

Then the code might be more readable as well..

Reply:
Ohh, yeah. I'll do that